wheel/src/adapt_response.rs

```diff
@@ -1,39 +1,26 @@
+use std::rc::Rc;
+
 use crate::lazy_node::LazyNode;
 use clvmr::allocator::Allocator;
-use clvmr::reduction::{EvalErr, Response};
-
-use std::rc::Rc;
+use clvmr::reduction::Response;
 
 use pyo3::prelude::*;
-use pyo3::types::PyDict;
 
-pub fn eval_err_to_pyresult<T>(py: Python, eval_err: EvalErr, allocator: Allocator) -> PyResult<T> {
-    let node = LazyNode::new(Rc::new(allocator), eval_err.0);
-    let msg = eval_err.1;
-    let ctx: &PyDict = PyDict::new(py);
-    ctx.set_item("msg", msg)?;
-    ctx.set_item("node", node)?;
-    Err(py
-        .run(
-            "
-from clvm.EvalError import EvalError
-raise EvalError(msg, node)",
-            None,
-            Some(ctx),
-        )
-        .unwrap_err())
-}
-
-pub fn adapt_response_to_py(
+pub fn adapt_response(
     py: Python,
     allocator: Allocator,
-    r: Response,
-) -> PyResult<(u64, LazyNode)> {
-    match r {
+    response: Response,
+) -> PyResult<(PyObject, LazyNode)> {
```

Comment:
why is this returning a `PyObject` now, instead of `u64`?

Reply:
The idea is to return an `int` on success or a `str` (containing a human-readable message) on failure. This eliminates the dependency on `EvalError` (which is implemented in python).

Reply:
ok. that still leaves a second form of failure where the `PyResult` is an `Err`, right?

Reply:
That's the `str` case. The python code will do a switch on the type returned to determine success vs failure, and then turns it into an `EvalError` on failure. This api is not expected to be used without the python wrapper.

Having a hybrid wheel means stuff that doesn't need to be fast and can be written in python when it's easier to do so. This wrapping code (which would be ugly in rust) is an example.

Reply:
The tests and benchmarks were failing, and to fix them, I had to write the boilerplate I mentioned above, so you can now see first hand how it works.